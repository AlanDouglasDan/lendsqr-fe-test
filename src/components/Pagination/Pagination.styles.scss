.rowsPerPageBtn {
  font-size: 14px;
  color: #545f7d;
  font-weight: 500;
  background-color: rgba(33, 63, 125, 0.1);
  padding: 0 12px;
  height: 30px;
  border-radius: 4px;
  border: none;
}

.operandButton {
  background-color: rgba(33, 63, 125, 0.1);
  border-radius: 4px;
  display: flex;
  align-items: center;
  justify-content: center;
  width: 24px;
  height: 24px;
  cursor: pointer;
  border: none;
}

.paginationNavigator {
  color: #545f7d;
  margin: 0 10px;
  opacity: 0.6;
  cursor: pointer;
}

.activePaginationNavigator {
  opacity: 1;
}
