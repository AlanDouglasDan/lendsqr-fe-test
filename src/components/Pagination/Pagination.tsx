import { getPaginatedPages } from "../../utils";
import "./Pagination.styles.scss";

type Props = {
  rowsPerPage: number;
  setRowsPerPage: any;
  numRecords: string;
  page: number;
  setPage: any;
  numberOfPages: number;
};

const Pagination: React.FC<Props> = ({
  rowsPerPage,
  setRowsPerPage,
  numRecords,
  page,
  setPage,
  numberOfPages,
}) => {
  const _pagesArray = getPaginatedPages(page, numberOfPages);

  // console.log(_pagesArray);

  return (
    <div className="d-flex align-items-center justify-content-between mt-20px center-grid">
      <div className="d-flex align-items-center">
        <div>Showing</div>
        <div className="dropdown">
          <button
            className="dropdown-toggle mx-2 rowsPerPageBtn"
            type="button"
            data-bs-toggle="dropdown"
            aria-expanded="false"
          >
            {rowsPerPage}
            <i className="fa fa-angle-down ms-2" />
          </button>
          <ul className="dropdown-menu">
            <li className="dropdown-item" onClick={() => setRowsPerPage(5)}>
              5
            </li>
            <li className="dropdown-item" onClick={() => setRowsPerPage(10)}>
              10
            </li>
            <li className="dropdown-item" onClick={() => setRowsPerPage(20)}>
              20
            </li>
          </ul>
        </div>
        <div>out of {numRecords}</div>
      </div>

      <div className="d-flex align-items-center">
        <button
          className="operandButton me-3"
          onClick={() => setPage(page - 1)}
          disabled={page === 1}
        >
          <i className="fa fa-angle-left" />
        </button>

        {_pagesArray.map((_page, index) => {
          if (_page === -1) return <div key={index}>...</div>;
          else
            return (
              <div
                className={`paginationNavigator ${
                  page === _page && "activePaginationNavigator"
                }`}
                onClick={() => setPage(_page)}
                key={index}
              >
                {_page}
              </div>
            );
        })}

        <button
          className="operandButton ms-3"
          onClick={() => setPage(page + 1)}
          disabled={page === numberOfPages}
        >
          <i className="fa fa-angle-right" />
        </button>
      </div>
    </div>
  );
};

export default Pagination;
