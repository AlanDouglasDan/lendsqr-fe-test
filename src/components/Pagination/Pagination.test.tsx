import { BrowserRouter } from "react-router-dom";
import { render } from "@testing-library/react";

import Pagination from "./Pagination";

describe("Pagination Component Test", () => {
  it("Pagination Component Snapshot", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Pagination
          rowsPerPage={10}
          setRowsPerPage={() => {}}
          numRecords="100"
          page={1}
          setPage={() => {}}
          numberOfPages={1}
        />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
