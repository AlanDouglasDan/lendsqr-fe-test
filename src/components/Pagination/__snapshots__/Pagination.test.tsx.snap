// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Component Test Pagination Component Snapshot 1`] = `
<DocumentFragment>
  <div
    class="d-flex align-items-center justify-content-between mt-20px center-grid"
  >
    <div
      class="d-flex align-items-center"
    >
      <div>
        Showing
      </div>
      <div
        class="dropdown"
      >
        <button
          aria-expanded="false"
          class="dropdown-toggle mx-2 rowsPerPageBtn"
          data-bs-toggle="dropdown"
          type="button"
        >
          10
          <i
            class="fa fa-angle-down ms-2"
          />
        </button>
        <ul
          class="dropdown-menu"
        >
          <li
            class="dropdown-item"
          >
            5
          </li>
          <li
            class="dropdown-item"
          >
            10
          </li>
          <li
            class="dropdown-item"
          >
            20
          </li>
        </ul>
      </div>
      <div>
        out of 100
      </div>
    </div>
    <div
      class="d-flex align-items-center"
    >
      <button
        class="operandButton me-3"
        disabled=""
      >
        <i
          class="fa fa-angle-left"
        />
      </button>
      <div
        class="paginationNavigator activePaginationNavigator"
      >
        1
      </div>
      <div>
        ...
      </div>
      <div
        class="paginationNavigator false"
      >
        0
      </div>
      <div
        class="paginationNavigator activePaginationNavigator"
      >
        1
      </div>
      <button
        class="operandButton ms-3"
        disabled=""
      >
        <i
          class="fa fa-angle-right"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
